src/Digdir.Domain.Dialogporten.Application/Features/V1/ServiceOwner/Dialogs/Queries/Search/SearchDialogQuery.cs

```diff
@@ -169,6 +173,7 @@ public async Task<SearchDialogResult> Handle(SearchDialogQuery request, Cancella
         var paginatedList = await dialogQuery
             .Include(x => x.Content)
             .ThenInclude(x => x.Value.Localizations)
+            .WhereIf(request.IdempotentKey is not null, x => x.IdempotentKey == request.IdempotentKey)
```

Comment:
I'm not sure if it makes sense to offer this as a query parameter; don't we want consumers to handle this optimistically via the 409 Conflict? 

In any case, this doesn't take into account the `org` and thus leaks the existence of used idempotency keys across service owners.